### PR TITLE
Move Druid release artifacts to Sonatype

### PR DESCRIPTION
**Pull request by xvrl**
- Switch to using Druid parent POM
- Add required fields for Sonatype
- Common plugin versions and settings have been moved to the parent pom
- Cleanup artifacts and POMs for consistent formatting
- Remove `org.hyperic.sigar` dependency and update docs to reflect necessary jars to add at runtime when sigar is needed
- Removed Kafka 0.7 extension from default build, since there are no Maven artifacts for Kafka 0.7
- Upgrade to Kafka 0.8.2.0

### CODE REVIEW

**Comment by fjy** [comment]
+1! Travis is also failing @_@

**Comment by xvrl** [comment]
Looks like I need to make org.hyperic.sigar optional and give instructions as to how to include it, since it's not in maven central.

**Comment by fjy** [comment]
I thought we had made it optional to include as a metrics monitor. we should consider making sigar an extension?

**Comment by xvrl** [comment]
Need to update server-metrics first

**Comment by xvrl** [comment]
It is already an "extension", in the sense that you can choose which monitors to include at runtime.
No need to move it into a separate module.

File: aws-common/pom.xml [hunk 1]
```diff
@@ -16,63 +16,37 @@
...
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-  <modelVersion>4.0.0</modelVersion>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
 
-  <artifactId>druid-aws-common</artifactId>
-  <name>druid-aws-common</name>
-  <description>druid-aws-common</description>
+    <artifactId>druid-aws-common</artifactId>
```
Comment (drcrallen) [inline aws-common/pom.xml]: Other extensions have groupId defined, is there a reason this one doesn't?

File: common/pom.xml [hunk 2]
```diff
@@ -16,9 +16,10 @@
   ~ limitations under the License.
   -->
 
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
-    <groupId>io.druid</groupId>
```
Comment (drcrallen) [inline common/pom.xml]: Why is this being removed?

File: extensions/hdfs-storage/pom.xml [hunk 3]
```diff
@@ -16,86 +16,61 @@
...
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
 
-  <build>
-    <plugins>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-source-plugin</artifactId>
```
Comment (drcrallen) [inline extensions/hdfs-storage/pom.xml]: With these all gone, does it actually include the sources when it builds?

File: common/pom.xml (hunk 2 above)
Comment (xvrl) [inline common/pom.xml]: groupId is redundant, since it is already specified by the parent pom

File: extensions/hdfs-storage/pom.xml (hunk 3 above)
Comment (xvrl) [inline extensions/hdfs-storage/pom.xml]: this was all moved to the parent pom as part of the release profile, see https://github.com/druid-io/oss-parent/blob/master/pom.xml

**Comment by xvrl** [comment]
Need to remove dependency on kafka-7 which does't have a maven central artifact. Anyone have any objections?

**Pull request merged**